src/libsyntax/parse/parser.rs

```diff
@@ -6172,6 +6174,45 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_expansion(&mut self, kind: ExpansionKind) -> PResult<'a, Expansion> {
```

Comment:
I agree that we should minimize adding stuff to the parser in general, but I think this is a simple and small enough function with similar enough structure to the other `parse_*` functions to warrant being in `parser.rs`, but I don't feel too strongly -- I'd be OK moving it to `expand.rs`.